fairing/preprocessors/base.py

```diff
@@ -61,9 +61,13 @@ def context_map(self):
 
     def context_tar_gz(self, output_file=constants.DEFAULT_CONTEXT_FILENAME):
         self.input_files = self.preprocess()
-        with tarfile.open(output_file, "w:gz") as tar:
+        added_files = set()
```

Comment:
I see it is used in line 68.

Reply:
ack

Reply:
Yes, I found a quite a few places where duplicate files can be added to the tarfile, so this is a check to make sure that duplicates are skipped when we are creating the context package itself.

The context map also means that it is possible for multiple input files to map to a single file in the archive, so we store the set of filenames that have actually been added.

Reply:
Can we move this to be a set in the init function? It's a bit weird to have it be a map, but enforce set qualities later on.

Reply:
That sounds reasonable, so we can essentially merge `context_map` and `added_files` into a single set. This will also help to keep the source / destination filenames consistent.

Reply:
Updated to reverse the context map direction to avoid duplicates (destination -> source file), and also add warning messages when duplicate files are skipped.